Added ListNotifier tests, removed ManualNotifier

dancamdev opened the pull request:
I've added some ListNotifier tests, and I removed the ManualNotifier which got replaced by the ChangeNotifier mixin in the ListNotifier class.

dancamdev committed:
removed unused manual notifier

dancamdev committed:
removed generic test file

dancamdev committed:
fixed docs misspelling

dancamdev committed:
added tests for list notifier

escamoteur commented on test/list_notifier_test.dart:
```diff
@@ -0,0 +1,157 @@
+    list.swap(0, 2);
+
+    expect(list.value, [3, 2, 1]);
+  });
```
Could you add a check that the notifyListener is called here too?

dancamdev replied:
Sure, will do it later today

escamoteur commented on test/list_notifier_test.dart:
```diff
@@ -0,0 +1,157 @@
+  });
+
+  test("Listeners get notified if an iterable is added", () {
+    list.addListener(() {
```
Sorry to correct you here.
if the Listener isn't called `expect` won't be called ever.
So the right way to do it is to assign some value inside the handler and check that this assignment with an expect

escamoteur replied:
this applies to all the following tests

dancamdev replied:
No, you are perfectly right, my bad, I'll fix it later today

dancamdev replied:
I bumped into a problem, the listener isn't called immediately, therefore the following test will fail because result is empty. Do you have any suggestion?

```
test("Listeners get notified on swap", () async {
    list.addListener(() {
      result = list.value;
    });

    list.swap(0, 2);

    expect(result, [3, 2, 1]);
  });
```

escamoteur replied:
I just tried this here:
```Dart
  ListNotifier list;

  setUp(() {
    list = ListNotifier(data: [1, 2, 3]);
  });
```
and it passed the test without any problems

escamoteur commented on test/list_notifier_test.dart:
```diff
@@ -0,0 +1,157 @@
+
+    list.sort((value1, value2) => - (value1.compareTo(value2)));
+  });
+}
```
I miss a test for the equality flag and the equality handler that you can pass to the constructor.
Also a test for the `[]`access to a list element. where that equality is used.
you could put this with its setup in a test group

escamoteur reviewed (commented):
Many thanks for your work! If you could address the points I commented it would be awesome.

dancamdev committed:
updated tests

dancamdev commented:
I've updated the tests as you suggested above. expect isn't called in the listener anymore and I've added the equality tests. Let me know if it's good for you!

escamoteur commented:
So the Test now works? do you have an idea what was wrong?

escamoteur commented:
Why did you use `addAll()` in the tests and not just assign the array like?

```Dart
  test("Listeners get notified on swap", () async {
    dynamic result;
    list.addListener(() {
      result = list.value;
    });

    list.swap(0, 2);

    expect(result, [3, 2, 1]);
  });
```

dancamdev commented:
Yes, I had a variable naming conflict I didn't see 🙃. It was late, I was tired 😅
